Assets/Packages/com.studio23.ss2.dialoguesystem/Runtime/Core/DialogueUI.cs

```diff
@@ -0,0 +1,111 @@
+
+using Cysharp.Threading.Tasks;
+using Studio23.SS2.DialogueSystem.Core;
+using Studio23.SS2.DialogueSystem.Data;
+using System.Threading;
+using TMPro;
+using UnityEngine;
+using UnityEngine.UI;
+
+
+namespace Studio23.SS2.DialogueSystem.UI
+{
+
+    public class DialogueUI : MonoBehaviour
+    {
+        [Header("UI Components")]
+        public TextMeshProUGUI dialogueText;
+        public Image BackgroundImage;
+        public GameObject UIRoot;
+        public Button NextButton;
+        public Button SkipButton;
+
+        private DialogueGraph _currentGraph;
+        [SerializeField]private DialogueState dialogueState;
+
+        [Header("Configuration")]
+        [SerializeField]private DialogueUIConfig config;
+
+
+
+      
+
+        private CancellationTokenSource CancelCurrentDialogueLineToken;
+
+
+        void Start()
+        {
+            NextButton.onClick.AddListener(Next);
+            ApplyConfiguration();
+        }
+
+        private void ApplyConfiguration()
+        {
+            dialogueText.fontSize = config.subtitleFontSize;
+            dialogueText.color = config.subtitleColor;
+            BackgroundImage.gameObject.SetActive(config.ShowBackground);
+            BackgroundImage.color = config.BackGroundColor;
+        }
+
+
+        public void RegisterGraph(DialogueGraph newGraph)
+        {
+            _currentGraph = newGraph;
+            _currentGraph.ClearEvents();
+            _currentGraph.OnDialogueStart += () => PlayDialogue();
+            _currentGraph.OnDialogueComplete += () => EndDialogue();
+            _currentGraph.OnDialogueNext += async () => await ShowDialogueText(_currentGraph.CurrentNode.DialogueText);
```

Comment:
Lacks readability considering everyone is still not familiar with lambda expression short-hands mixed with async function calls.